product_supplierinfo_for_customer_sale/models/sale_order_line.py

```diff
@@ -0,0 +1,63 @@
+# Copyright 2013-2017 Agile Business Group sagl
+#     (<http://www.agilebg.com>)
+# Copyright 2021 ForgeFlow S.L. (https://www.forgeflow.com)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    product_customer_code = fields.Char(
+        compute="_compute_product_customer_code",
+        string="Product Customer Code",
+    )
+
+    @api.depends("product_id")
+    def _compute_product_customer_code(self):
+        for line in self:
+            if line.product_id:
+                supplierinfo = line.product_id._select_customerinfo(
+                    partner=line.order_partner_id
+                )
+                code = supplierinfo.product_code
+            else:
+                code = ""
+            line.product_customer_code = code
+
+    @api.onchange("product_id")
+    def product_id_change(self):
+        result = super(SaleOrderLine, self).product_id_change()
+        for line in self.filtered(
+            lambda sol: sol.product_id.product_tmpl_id.customer_ids
+            and sol.order_id.pricelist_id.item_ids
+        ):
+            product = line.product_id
+            items = self.env["product.pricelist.item"].search(
```

Comment:
@LoisRForgeFlow Would it be possible to do the search outside the loop ?

Reply:
It is an onchange method, 99% of the times will be called with one line. So, I would prefer to stick to a strict migration and do not include that change.